src/completion.rs

```diff
@@ -604,10 +604,11 @@ fn get_stdlib_matches(
     let mut matches = vec![];
     let completes = get_stdlib_completables();
 
-    for c in completes.into_iter() {
-        if c.matches(name.clone(), info.clone()) {
-            matches.push(c.completion_item(info.clone()));
-        }
+    for c in completes
+        .into_iter()
+        .filter(|x| x.matches(name.clone(), info.clone()))
```

Comment:
It seems like `matches` requiring a `String` could impact performance, and that `&str` would be better?

Reply:
Yeah, like above, the `completion` module is ripe for refactor. I changed `Completable.matches` to take borrows of the `name` and the `info`, and discovered that the `name` was just using the `str` representation down the line, so might as well not `.clone()` anyway.